pkg/generate/code/resource_reference.go

```diff
@@ -179,6 +185,187 @@ func ReferenceFieldsPresent(
+			outPrefix += resolveSingleReference(field, fieldAccessPrefix, refTarget, false, indentLevel+idx)
+			outPrefix += fmt.Sprintf("%s%s = &%s\n", indent, targetVarName, refTarget)
+		}
+	}
+
+	return outPrefix + outSuffix
+}
+
+func resolveSingleReference(field *model.Field, sourceVarName string, targetVarName string, shouldAppendTarget bool, indentLevel int) string {
+	out := ""
+	indent := strings.Repeat("\t", indentLevel)
+
+	out += fmt.Sprintf("%sif %s != nil && %s.From != nil {\n", indent, sourceVarName, sourceVarName)
+	out += fmt.Sprintf("%s\tarr := %s.From\n", indent, sourceVarName)
+	out += fmt.Sprintf("%s\tif arr == nil || arr.Name == nil || *arr.Name == \"\" {\n", indent)
+	out += fmt.Sprintf("%s\t\treturn fmt.Errorf(\"provided resource reference is nil or empty\")\n", indent)
+	out += fmt.Sprintf("%s\t}\n", indent)
+	out += fmt.Sprintf("%s\tnamespacedName := types.NamespacedName{\n", indent)
+	out += fmt.Sprintf("%s\t\tNamespace: namespace,\n", indent)
+	out += fmt.Sprintf("%s\t\tName: *arr.Name,\n", indent)
+	out += fmt.Sprintf("%s\t}\n", indent)
+	if field.FieldConfig.References.ServiceName == "" {
+		out += fmt.Sprintf("%s\tobj := svcapitypes.%s{}\n", indent, field.FieldConfig.References.Resource)
+	} else {
+		out += fmt.Sprintf("%s\tobj := %sapitypes.%s{}\n", indent, field.ReferencedServiceName(), field.FieldConfig.References.Resource)
+	}
+	out += fmt.Sprintf("%s\terr := apiReader.Get(ctx, namespacedName, &obj)\n", indent)
+	out += fmt.Sprintf("%s\tif err != nil {\n", indent)
+	out += fmt.Sprintf("%s\t\treturn err\n", indent)
+	out += fmt.Sprintf("%s\t}\n", indent)
+	out += fmt.Sprintf("%s\tvar refResourceSynced, refResourceTerminal bool\n", indent)
+	out += fmt.Sprintf("%s\tfor _, cond := range obj.Status.Conditions {\n", indent)
+	out += fmt.Sprintf("%s\t\tif cond.Type == ackv1alpha1.ConditionTypeResourceSynced &&\n", indent)
+	out += fmt.Sprintf("%s\t\t\tcond.Status == corev1.ConditionTrue {\n", indent)
+	out += fmt.Sprintf("%s\t\t\trefResourceSynced = true\n", indent)
+	out += fmt.Sprintf("%s\t\t}\n", indent)
+	out += fmt.Sprintf("%s\t\tif cond.Type == ackv1alpha1.ConditionTypeTerminal &&\n", indent)
+	out += fmt.Sprintf("%s\t\t\tcond.Status == corev1.ConditionTrue {\n", indent)
+	out += fmt.Sprintf("%s\t\t\trefResourceTerminal = true\n", indent)
+	out += fmt.Sprintf("%s\t\t}\n", indent)
+	out += fmt.Sprintf("%s\t}\n", indent)
+	out += fmt.Sprintf("%s\tif refResourceTerminal {\n", indent)
+	out += fmt.Sprintf("%s\t\treturn ackerr.ResourceReferenceTerminalFor(\n", indent)
+	out += fmt.Sprintf("%s\t\t\t\"%s\",\n", indent, field.FieldConfig.References.Resource)
+	out += fmt.Sprintf("%s\t\t\tnamespace, *arr.Name)\n", indent)
+	out += fmt.Sprintf("%s\t}\n", indent)
+	out += fmt.Sprintf("%s\tif !refResourceSynced {\n", indent)
+	out += fmt.Sprintf("%s\t\treturn ackerr.ResourceReferenceNotSyncedFor(\n", indent)
+	out += fmt.Sprintf("%s\t\t\t\"%s\",\n", indent, field.FieldConfig.References.Resource)
```

Comment:
I see that you have mutiple instances of `\"%s\"`, you can also use `%q` to format a string + adding double quotes around it